src/oop/object_interactions/medium/q4/QuestionText.md

```diff
@@ -0,0 +1,46 @@
+# oop-object_interactions-medium-q4
+
+- Java question
+
+## Snail Class
+
+The `Snail` class currently has the following items:
+
+- Attributes:
+    - A `char` called **bloodType**
+    - A `double` called **volume**
+    - None of these attributes should be accessible from other classes.
+
+- Constructors:
+    - A constructor that take a `char` and a `double` to initialize the **bloodType** and **volume** attributes.
+    - It is accessible from other classes.
+
+- Methods:
+    - Getter and Setter methods for **bloodType** and **volume**.
+    - These methods are accessible from other classes.
+
+You must add the following items to the class:
+
+- Methods:
+    - A `toString` method that prints the object information in the format
+      `My blood type is x and I have yL of blood left` where x is the **bloodType** value and y is the **volume** value.
+    - A `transfusionFrom` method that take another **Snail** object to do a blood tansfer.
+    - It should check if the blood types are a match and if not print `Blood type incompatible`. Note that in this scenario
+  **Snail** blood types are not like humans, so you should check that the type match exactly.
+    - If the types are compatible it should say `Checking if donor has enough blood for a transfusion...`
+    - Then it should make sure that the donor has more than 5 unit of blood if not
+      it should print `Donor does not have enough blood for a transfusion`.
+    - If donor has enough take 1 unit of blood from donor
+      and give it to the snail, and then print `Blood transfusion completed`.
```

Comment:
These need to be tabbed over, they are not currently sub-bullets under the `transfusion` method.

Reply:
Also, specify if all the print statements should be on different lines.